controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -1186,6 +1220,13 @@ protected ReaderGroup getReaderGroup(String scope, final String name, OperationC
         return Futures.completeOn(checkScopeExists(scope, context, executor)
                    .thenCompose(exists -> {
                    if (exists) {
+                       checkScopeInDeletingTable(scope, context, executor)
+                               .thenCompose(exist -> {
+                                   if (exist) {
+                                       throw new IllegalArgumentException("Scope already in deleting state: " + scope);
```

Comment:
Again we should not be throwing IllegalArgumentException if scope is already in deleting state.

Reply:
exception is changed here too